### PR TITLE
Chat Subsystem ported from TerraGov-Marine-Corps

### PR DESCRIPTION
**Pull request by tralezab**
Really all credit goes to @DominikPanic for the code, I just made a pr here.

## About The Pull Request

This is just a subsystem that queues to_chats, so they cannot lag out your game. If the subsystem doesn't work, it will just revert back to the old system.

I really gotta test it more, because i'm not perfectly sure everything is alright yet!

## Why It's Good For The Game

https://github.com/tgstation/TerraGov-Marine-Corps/issues/746
THIS COULD BE YOU!

### CODE REVIEW

**Comment by TerraGS** [comment]
Low key shotgun nerf

**Comment by optimumtact** [comment]
it's not going to fix the lag but it will help

File: code/modules/goonchat/browserOutput.dm [hunk 1]
```diff
@@ -258,6 +258,12 @@ GLOBAL_DATUM_INIT(iconCache, /savefile, new("tmp/iconCache.sav")) //Cache of ico
 		// url_encode it TWICE, this way any UTF-8 characters are able to be decoded by the Javascript.
 		C << output(url_encode(url_encode(message)), "browseroutput:output")
 
+/proc/to_chat(target, message, handle_whitespace = TRUE)
+	if(!SSchat)
+		to_chat_immediate(target, message, handle_whitespace)
```
Comment (vuonojenmustaturska) [inline code/modules/goonchat/browserOutput.dm]: this path will never run unless someone finds a new sdql exploit to harddel the subsystem

Comment (DominikPanic) [reply code/modules/goonchat/browserOutput.dm]: `if(Master.current_runlevel == RUNLEVEL_INIT || !SSchat?.initialized)` this is the current if check here, you should update the PR @oranges as there were some important QoL fixes with loading

Comment (tralezab) [reply code/modules/goonchat/browserOutput.dm]: 👌 , thank you

File: code/controllers/subsystem/chat.dm [hunk 2]
```diff
@@ -0,0 +1,67 @@
+SUBSYSTEM_DEF(chat)
+	name = "Chat"
+	flags = SS_TICKER|SS_NO_INIT
+	wait = 1
+	priority = FIRE_PRIORITY_CHAT
+
```
Comment (DominikPanic) [inline code/controllers/subsystem/chat.dm]: You also definitely want it to init for reasons mentioned here https://github.com/tgstation/TerraGov-Marine-Corps/pull/2134 , if you want to be safe you should be able to just copy the lastest version of that file from our  codebase.

**Comment by tralezab** [comment]
Let me add that before this is tried again

File: code/controllers/subsystem/chat.dm (hunk 2 above)
Comment (tralezab) [reply code/controllers/subsystem/chat.dm]: added

**Comment by actioninja** [comment]
This isn't perfect but it's a massive improvement.

**Comment by optimumtact** [comment]
this was test merged yes?

**Comment by tralezab** [comment]
Yep. No big issues jumped up as far as I'm aware.

**Pull request merged**